tests/e2e/e2e_ignore_namespace_test.go

```diff
@@ -0,0 +1,197 @@
+package e2e
+
+import (
+	"context"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/openservicemesh/osm/pkg/constants"
+	. "github.com/openservicemesh/osm/tests/framework"
+)
+
+var _ = OSMDescribe("Ignore Namespaces",
+	OSMDescribeInfo{
+		Tier:   1,
+		Bucket: 2,
+	},
+	func() {
+		Context("Ignore Label", func() {
+			const ignoreNs = "ignore"
+			const monitorIgnoreNs = "mesh-ignore"
+			const sidecarMonitorIgnoreNs = "sidecar-monitor-ignore"
+			var ns []string = []string{ignoreNs, monitorIgnoreNs, sidecarMonitorIgnoreNs}
+
+			It("Tests the ignore label on a namespace disables sidecar injection", func() {
+				// Install OSM
+				installOpts := Td.GetOSMInstallOpts()
+				installOpts.EnablePermissiveMode = true
+				Expect(Td.InstallOSM(installOpts)).To(Succeed())
+
+				// Create test NS in mesh with ignore label
+				for _, n := range ns {
+					Expect(Td.CreateNs(n, map[string]string{constants.IgnoreLabel: "true"})).To(Succeed())
+				}
+
+				// Add monitor-ignore to mesh with sidecar injection disabled
+				Expect(Td.AddNsToMesh(false, monitorIgnoreNs)).To(Succeed())
```

Comment:
This test case here seems unnecessary given that this test is meant to test the ignore label. Just testing ignore label with sidecar-injection enabled should suffice. This is a k8s functionality that we are testing now and not OSM, so I suggest simplifying this test to only test the ignore label and not all combinations of the `matchExpression` selector exposed by the k8s API.